dt-posts/dt-posts.php

```diff
@@ -1340,7 +1345,551 @@ public static function get_post_activity( string $post_type, int $post_id, array
+                    break;
+                case 'number':
+                    $number_update_allowed = true;
+
+                    // Ensure to adhere with any min/max bounds, to avoid exceptions!
+                    $number = !empty( $reverted['values'][0] ) ? $reverted['values'][0] : 0;
+                    if ( isset( $post_type_fields[$field_key], $post_type_fields[$field_key]['min_option'] ) && $post_type_fields[$field_key]['min_option'] > $number ){
+                        $number_update_allowed = false;
+                    }
+                    if ( isset( $post_type_fields[$field_key], $post_type_fields[$field_key]['max_option'] ) && $post_type_fields[$field_key]['max_option'] < $number ){
+                        $number_update_allowed = false;
+                    }
+
+                    // Only update if number format is valid.
+                    if ( $number_update_allowed ){
+                        $post_updates[$field_key] = $number;
+                    }
+                    break;
+                case 'user_select':
+                    $user_select_value = $reverted['values'][0];
+                    if ( $user_select_value != 'user-' ){
+                        $post_updates[$field_key] = $user_select_value;
+                    } else {
+                        $post_updates[$field_key] = '';
+                    }
+                    break;
+                case 'text':
+                case 'boolean':
+                case 'textarea':
+                case 'key_select':
+                    $post_updates[$field_key] = $reverted['values'][0] ?? '';
+                    break;
+            }
+        }
+
+        /**
+         * Final home-straight - Submit updates to revert post back to specified state.
+         * Simply return blank, if no updates are to be made.
+         */
+
+        if ( empty( $post_updates ) ){
+            return [];
+        }
+
+        // Ensure revert activity carried out by Revert Bot, with sufficient permissions.
+        $current_user_id = get_current_user_id();
+        wp_set_current_user( 0 );
+        $revert_bot = wp_get_current_user();
+        $revert_bot->add_cap( 'activity_revert' );
+        $revert_bot->add_cap( 'dt_all_access_contacts' );
+        $revert_bot->add_cap( 'update_any_contacts' );
+        $revert_bot->add_cap( 'dt_all_access_groups' );
+        $revert_bot->add_cap( 'update_any_groups' );
+        $revert_bot->add_cap( 'dt_all_access_peoplegroups' );
+        $revert_bot->add_cap( 'update_any_peoplegroups' );
+
+        if ( !in_array( $post_type, [ 'contacts', 'groups', 'peoplegroups' ] ) ){
+            $revert_bot->add_cap( 'dt_all_access_' . $post_type );
+            $revert_bot->add_cap( 'update_any_' . $post_type );
+        }
+
+        $revert_bot->display_name = __( 'Revert Bot', 'disciple_tools' );
+
+        // Update post based on reverted values.
+        $updated_post = self::update_post( $post_type, $post_id, $post_updates );
```

Comment:
@kodinkat, we could use check_permissions = false, instead of adding "all" capabilities.
This should also work for other post types beyond contacts, groups and ppl groups